### PR TITLE
Add new custom hlint rule for runSetting.

### PR DESCRIPTION
**Pull request by elland**
Also applies hlint again to the whole codebase (excluding tests), as we had some drift between finalising hlint and new PRs being merged without being linted / having CI catch those cases.

I also disabled the pipefail from the script, as that would short-circuit the linter on first issue found. Hopefully that doesn't mess with CI.

PS: This will fail CI linters phase until #2715 has been merged.

### CODE REVIEW

File: tools/hlint.sh [hunk 1]
```diff
@@ -1,7 +1,5 @@
 #!/usr/bin/env bash
 
-set -euo pipefail
-
```
Comment (akshaymankar) [inline tools/hlint.sh:4]: Why remove this?

**Review by akshaymankar — COMMENTED** [review]
Does checking only files changed by PR mean that adding new rules won't run over everything in CI? Maybe we should reconsider this approach and just run hlint on everything.

File: tools/hlint.sh (hunk 1 above)
Comment (elland) [reply tools/hlint.sh:4]: From the PR description:

> I also disalbed the pipefail from the script, as that would short-circuit the linter on first issue found. Hopefully that doesn't mess with CI.

Does that make sense?

**Comment by elland** [comment]
@akshaymankar We might want to do that indeed, at least on CI. For this case, or when I touch the rules, I do run on everything (except tests), but that's a fraught process that requires people to remember to do manual steps, and as you mentioned somewhere else, linting runs parallel to the other CI steps, so it taking 5x or more longer wouldn't impact us, as it would remain much faster than integration tests. I'll bring this up to the rest of the team.

File: tools/hlint.sh (hunk 1 above)
Comment (akshaymankar) [reply tools/hlint.sh:4]: Ah, sorry I didn't read the description! I am sure there is way we can find, but I don't have much time to spend on this, so I will take back my objection.

**Review by akshaymankar — APPROVED** [review]

**Pull request merged**